### PR TITLE
CODE-209: Fix-up move menu history from recipe book to correct service

### PR DESCRIPTION
**Pull request by AkagiShigeru**
* process all hard-coded recipes first
* if a recipe is listed more than once, toss quality error
* prevent hard-coded & already selected random recipes from being selected again by using exlude_uuid_list

### CODE REVIEW

File: sous-chef/sous_chef/menu/create_menu.py [hunk 1]
```diff
@@ -550,8 +601,15 @@ def _retrieve_manual_menu_ingredient(
         )
 
     @BaseWithExceptionHandling.ExceptionHandler.handle_exception
-    def _retrieve_recipe(self, row: pd.Series):
+    def _retrieve_recipe(self, row: pd.Series, processed_uuid_list: List):
         recipe = self.recipe_book.get_recipe_by_title(row["item"])
+        if row.override_check == "N" and recipe.uuid in processed_uuid_list:
```
Comment (ArBridgeman) [inline sous-chef/sous_chef/menu/create_menu.py:606]: Do we want to be able to override both cases?

File: sous-chef/sous_chef/recipe_book/read_recipe_book.py [hunk 2]
```diff
@@ -78,7 +74,6 @@ class SelectRandomRecipeError(DirectSearchError):
     message = "[select random recipe failed]"
 
 
-@extend_enum([MapMenuHistoryErrorToException])
```
Comment (ArBridgeman) [inline sous-chef/sous_chef/recipe_book/read_recipe_book.py:81]: Double -check that menu history is not still in recipe book related config

File: sous-chef/sous_chef/menu/create_menu.py [hunk 3]
```diff
@@ -203,9 +221,33 @@ def finalize_fixed_menu(self):
...
+        ).drop(columns=["process_order"])
         # validate schema & process menu
         self._validate_menu_schema()
-        self.dataframe = self.dataframe.apply(self._process_menu, axis=1)
+
+        holder_dataframe = pd.DataFrame()
+        processed_uuid_list = []
+        for _, entry in self.dataframe.iterrows():
+            processed_entry = self._process_menu(
+                entry, processed_uuid_list=processed_uuid_list
```
Comment (ArBridgeman) [inline sous-chef/sous_chef/menu/create_menu.py:238]: Does it help/change any warnings if entry.copy() is used instead?

**Review by ArBridgeman — CHANGES_REQUESTED** [review]

File: sous-chef/tests/integration_tests/data/final_grocery_list.csv [hunk 4]
```diff
@@ -1,49 +1,48 @@
...
+cup,cup,shredded gratin cheese,False,0.25,True,Dairy products,shredded gratin cheese,grocery store,5760466896467,[Lasagna],2022-01-25 17:15:00+00:00,[Tue],False,Dairy force
+cup,cup,sour cream,False,0.3,True,Dairy products,sour cream,grocery store,4046700014301,[Holiday cheesecake],2022-01-23 11:30:00+00:00,[Sun],False,Dairy force
+cup,cup,sugar,False,0.5,True,Baking,sugar,grocery store,4006544205006,[Holiday cheesecake],2022-01-23 11:30:00+00:00,[Sun],False,Sweet carolina
+tsp,teaspoon,vanilla extract,False,2.0,True,Baking,vanilla extract,Homemade,,"[Baked Oatmeal Bars, Holiday cheesecake]",2022-01-23 11:30:00+00:00,[Sun],False,Homemade
+cup,cup,whey protein powder,False,0.25,True,Spices and herbs,whey protein powder,Amazon,683442920,[Baked Oatmeal Bars],2022-01-23 18:15:00+00:00,[Sun],False,Amazon
+can,can,whole tomatoes,False,1.0,True,Canned,whole tomatoes,grocery store,4311596062621,[Lasagna],2022-01-25 17:15:00+00:00,[Tue],False,Sauces cans and oils
+tbsp,tablespoon,yogurt,False,2.0,True,Dairy products,yogurt,grocery store,4062800016151,[Spanish Deluxe Potatoes],2022-01-23 18:15:00+00:00,[Sun],False,Dairy force
```
Comment (ArBridgeman) [inline sous-chef/tests/integration_tests/data/final_grocery_list.csv:48]: @AkagiShigeru any ideas why times changed? I.e. last entry went from 17:45 to 18:15. Wasn't the case in other or, was just a store change

**Review by ArBridgeman — CHANGES_REQUESTED** [review]

File: sous-chef/sous_chef/recipe_book/read_recipe_book.py (hunk 2 above)
Comment (ArBridgeman) [reply sous-chef/sous_chef/recipe_book/read_recipe_book.py:81]: not relevant as not handling exceptions so was never added here

File: sous-chef/pyproject.toml [hunk 5]
```diff
@@ -1,31 +1,30 @@
...
-google-auth-oauthlib = "^0.5"
 hydra-core = "^1.2"
-# future version of pandas removes append
-pandas = "^1.3.5"
-pandera = "^0.11.0"
+pandas = "^1.3"
+pandera = "^0.11"
 pint = "^0.19"
-pygsheets = "^2.0.5"
+# issue with test__get_workbook_with_folder with 2.0.6
```
Comment (ArBridgeman) [inline sous-chef/pyproject.toml:14]: @AkagiShigeru do you want to take a quick look or rather create a bug ticket to look at it another time?

File: sous-chef/tests/integration_tests/data/final_grocery_list.csv (hunk 4 above)
Comment (ArBridgeman) [reply sous-chef/tests/integration_tests/data/final_grocery_list.csv]: @AkagiShigeru maybe re-check out main. I didn't change anything except rebase off main, and this resolved itself.
see https://github.com/ArBridgeman/aurorianclouds/blob/CODE-209/sous-chef/tests/integration_tests/data/final_grocery_list.csv

File: sous-chef/pyproject.toml (hunk 5 above)
Comment (AkagiShigeru) [reply sous-chef/pyproject.toml:14]: will create bug ticket

**Review by ArBridgeman — APPROVED** [review]

**Pull request merged**